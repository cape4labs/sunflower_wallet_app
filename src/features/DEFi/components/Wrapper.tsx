import { View } from 'react-native';

type WrapperType = {
  children: React.ReactNode;
};

export function Wrapper({ children }: WrapperType) {
  return (
    <View className="border-2 border-red-500 h-5/6 w-5/6 m-auto justify-center flex-col">
      {children}
    </View>
  );
}
