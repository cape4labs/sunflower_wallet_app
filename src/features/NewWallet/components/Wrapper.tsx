import { ScrollView } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

type WrapperType = {
  children: React.ReactNode;
};

// For none scrollable pages
export function Wrapper({ children }: WrapperType) {
  return <SafeAreaView className="flex-1 items-center justify-center m-8">{children}</SafeAreaView>;
}

// For scrollable pages
export function ScrollableWrapper({ children }: WrapperType) {
  return (
    <ScrollView>
      <SafeAreaView className="flex-1 items-center justify-center m-8">{children}</SafeAreaView>
    </ScrollView>
  );
}
