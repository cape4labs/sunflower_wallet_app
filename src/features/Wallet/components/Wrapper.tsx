import { ScrollView, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

type WrapperType = {
  children: React.ReactNode;
};

export function Wrapper({ children }: WrapperType) {
  return (
    <SafeAreaView className='flex-1 items-center justify-center m-8'>
      {children}
    </SafeAreaView>
  );
}

export function ScrollableWrapper({ children }: WrapperType) {
  return(
    <ScrollView >
      <SafeAreaView className='flex-1 items-center justify-center m-8'>
        {children}
      </SafeAreaView>
    </ScrollView>
  )
}
