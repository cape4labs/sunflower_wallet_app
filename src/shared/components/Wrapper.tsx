import { SafeAreaView } from 'react-native-safe-area-context';

type WrapperType = {
  children: React.ReactNode;
};

// For none scrollable pages
export default function Wrapper({ children }: WrapperType) {
  return <SafeAreaView className="flex-1 items-center justify-center m-8">{children}</SafeAreaView>;
}
