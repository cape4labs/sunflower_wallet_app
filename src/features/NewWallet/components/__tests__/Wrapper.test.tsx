import { render } from '@testing-library/react-native';
import { Wrapper } from '../Wrapper';
import { Text } from 'react-native';

it('Wrapper snapshot', () => {
  const tree = render(Wrapper({ children: <Text /> })).toJSON();
  expect(tree).toMatchSnapshot();
});
