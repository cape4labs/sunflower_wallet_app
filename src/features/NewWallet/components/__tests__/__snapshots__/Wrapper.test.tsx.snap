// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper snapshot 1`] = `
<RNCSafeAreaView
  className="flex-1 items-center justify-center m-8"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <Text />
</RNCSafeAreaView>
`;
